tools/serverpod_cli/lib/src/analyzer/models/definitions.dart

```diff
@@ -253,6 +261,35 @@ class ProtocolEnumValueDefinition {
   ProtocolEnumValueDefinition(this.name, [this.documentation]);
 }
 
+abstract class InheritanceDefinition {
```

Comment:
Suggestion: If we simply store a reference to the class, do we really need BaseClass and ExtendsClass? Isn't everything we need available through the class definition? Then we could simple have an Unresolved and Resolved Inheritance definition.